### ISSUE TITLE
How to custom module label_names

### ISSUE THREAD

**Issue by cjfan** [apache/incubator-mxnet#2295]
I'm confused how to custom the label_names.
If I comment the line 'model.bind(...)'
the code will raise an error of not find the 'softmax_lable' (as the default name of lable),
But I have pass the label_names param in Module()
With uncomment the line 'model.bind',
the code will work but raise a warnning 'WARNING:root:Already binded, ignoring bind()'
So, what is the right way of set label_names and init the module?

```
label = mx.sym.Variable('label')
dloss  = mx.symbol.LogisticRegressionOutput(data = fc3, label=label, name = 'dloss')
model =  mx.mod.Module(symbol=dloss,data_names=('data',), label_names=('label',))
#model.bind( data_shapes=train_iter.provide_data,label_shapes=[('label', (batch_size,))])
model.fit(
    train_iter,
    eval_data=test_iter, 
    optimizer_params={'learning_rate':0.01, 'momentum': 0.9},
    num_epoch=10)
```

**Comment by antinucleon** [comment]
Please refer the example: https://github.com/dmlc/mxnet/tree/master/example/multi-task to customize data iterator.

**Comment by cjfan** [comment]
Thank you, and as mentioned in #866, after read the code in io.py, the implementation of NDArrayIter. I notice that the label_name can be dictionary... 
so if the code changed like:
```
train_iter = mx.io.NDArrayIter(XY_train[:,:2], {'label':XY_train[:,2]}, batch_size=batch_size)
test_iter = mx.io.NDArrayIter(XY_test[:,:2], {'label':XY_test[:,2]}, batch_size=batch_size)
```
it can work without call for bind.
I think when I want to set the label, I firstly consider the API from the Module, and a 'label_names' param is indeed here... 
Is there some reason that the assignment of label_names in Module not overwrite the value in DataIter?

**Comment by phunterlau** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!
